dbt/adapters/redshift/connections.py

```diff
@@ -38,33 +40,176 @@ class RedshiftConnectionMethod(StrEnum):
 
 
 @dataclass
-class RedshiftCredentials(PostgresCredentials):
+class RedshiftCredentials(Credentials):
+    host: str
+    user: str
+    port: Port
     method: str = RedshiftConnectionMethod.DATABASE  # type: ignore
     password: Optional[str] = None  # type: ignore
     cluster_id: Optional[str] = field(
         default=None,
         metadata={"description": "If using IAM auth, the name of the cluster"},
     )
     iam_profile: Optional[str] = None
-    iam_duration_seconds: int = 900
-    search_path: Optional[str] = None
-    keepalives_idle: int = 4
+    iam_duration_seconds: Optional[
+        int
+    ] = 900  # TODO:Not supported in redshift python connector, should we remove this?
+    search_path: Optional[
+        str
+    ] = None  # TODO:Not supported in redshift python connector, should we remove this?
+    keepalives_idle: int = (
+        4  # TODO:Not supported in redshift python connector, should we remove this?
+    )
     autocreate: bool = False
     db_groups: List[str] = field(default_factory=list)
-    ra3_node: Optional[bool] = False
+    ra3_node: Optional[
+        bool
+    ] = False  # TODO:Need to confirm if there is special handling needed for this option
+    connect_timeout: int = 10
+    role: Optional[str] = None
+    sslmode: Optional[str] = None
+    sslcert: Optional[
```

Comment:
@colin-rogers-dbt to follow-up on `sslcert`, `sslrootcert` and `sslkey` usage and if it's really needed to be kept.

Reply:
cc: @jtcohen6 &@Fleid 
Looking around I don't think this is being used (and I don't think it even makes sense for redshift anyways) so we can probably remove. AFAIK I don't think having extra values in the profile.yml will break anything but we should probably add a warning that it's no longer utilized?

Reply:
Not being used to my knowledge. Confirmed that users will not see an error if they add extra unused / unrecognized values into `profiles.yml`. I think a deprecation warning is reasonable, given that these were previously documented.